cpp/CMakeLists.txt

```diff
@@ -119,15 +128,26 @@ add_library(rapids_triton::rapids_triton ALIAS rapids_triton)
 target_include_directories(rapids_triton INTERFACE "$<BUILD_INTERFACE:${RAPIDS_TRITON_SOURCE_DIR}/include>"
                                           "$<INSTALL_INTERFACE:include>")
 
-target_link_libraries(rapids_triton
-INTERFACE
-  rmm::rmm
-  raft::raft
-  triton-core-serverstub
-  triton-backend-utils
+if(TRITON_ENABLE_GPU)
+  target_link_libraries(rapids_triton
+  INTERFACE
+    rmm::rmm
+    raft::raft
+    triton-core-serverstub
+    triton-backend-utils
   )
+else()
+  target_link_libraries(rapids_triton
+  INTERFACE
+    triton-core-serverstub
+    triton-backend-utils
+  )
+endif()
```

Comment:
Something like this would be slightly nicer

```cmake
target_link_libraries(rapids_triton
INTERFACE
  $<$<BOOL:${TRITON_ENABLE_GPU}>:rmm::rmm>
  $<$<BOOL:${TRITON_ENABLE_GPU}>:raft::raft>
  triton-core-serverstub
  triton-backend-utils
)
```